app/controllers/sitemaps_controller.rb

```diff
@@ -0,0 +1,22 @@
+class SitemapsController < ApplicationController
+  before_action :set_cache_control_headers, only: %i[show]
+
+  def show
+    date_string = params[:sitemap].gsub("sitemap-", "").gsub(".xml", "")
+    not_found unless date_string.match?(/\A[A-Z][a-z][a-z]\-\d{4}\z/)
```

Comment:
I think I'd pull the regex out into a constant and match against that. Also, we can use a named capture group to do all of this in one step:

```ruby
SITEMAP_REGEX = /\Asitemap-(?<date_string>[A-Z][a-z][a-z]-\d{4})\.xml\z/.freeze

def show
  match = "sitemap-Mar-200.xml".match(SITEMAP_REGEX)
  not_found unless match && match[:date_string]
  begin
    date = Time.zone.parse(match[:date_string]).at_beginning_of_month
  ...
end
```

if you don't like that, I'd still pull out the regex at least.

Reply:
Nice. Will do.